test/EventManagerIntrospectionTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Mvc;
+
+use ReflectionProperty;
+use Zend\EventManager\EventManager;
+
+/**
+ * Offer methods for introspecting event manager events and listeners.
+ */
+trait EventManagerIntrospectionTrait
+{
+    public function getEventsFromEventManager(EventManager $events)
```

Comment:
I know you hate traits, but I found it an expedient way to accomplish the
task, and this is precisely the dirt of thing they were designed for.

I'll update to private, though I don't think the visibility is important
within a unit test.
On Oct 1, 2015 6:43 AM, "Marco Pivetta" <notifications@github.com> wrote:

> In test/EventManagerIntrospectionTrait.php
> <https://github.com/zendframework/zend-mvc/pull/31#discussion_r40904132>:
>
> > + * @link      http://github.com/zendframework/zf2 for the canonical source repository
> > + * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
> > + * @license   http://framework.zend.com/license/new-bsd New BSD License
> > + */
> > +
> > +namespace ZendTest\Mvc;
> > +
> > +use ReflectionProperty;
> > +use Zend\EventManager\EventManager;
> > +
> > +/**
> > + * Offer methods for introspecting event manager events and listeners.
> > + */
> > +trait EventManagerIntrospectionTrait
> > +{
> > +    public function getEventsFromEventManager(EventManager $events)
>
> Why can't this be a static method somewhere (instead of a trait)? Also:
> private
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-mvc/pull/31/files#r40904132>.
>